### PR TITLE
Update packages to latest core

### PR DESCRIPTION
**Pull request by RickWinter**

### CODE REVIEW

File: sdk/messaging/azeventhubs/go.mod [hunk 1]
```diff
@@ -3,16 +3,16 @@ module github.com/Azure/azure-sdk-for-go/sdk/messaging/azeventhubs
 go 1.18
 
 require (
-	github.com/Azure/azure-sdk-for-go/sdk/azcore v1.4.0
+	github.com/Azure/azure-sdk-for-go/sdk/azcore v1.8.0
 	github.com/Azure/azure-sdk-for-go/sdk/azidentity v1.1.0
-	github.com/Azure/azure-sdk-for-go/sdk/internal v1.2.0
+	github.com/Azure/azure-sdk-for-go/sdk/internal v1.4.0
```
Comment (jhendrixMSFT) [inline sdk/messaging/azeventhubs/go.mod:8]: Will need to revert to `v1.3.0` until Ben gets the tests updated.

Comment (RickWinter) [reply sdk/messaging/azeventhubs/go.mod:8]: Made the fix to address this.  See updated change

**Pull request merged**